products/BellHybrid/apps/common/src/layouts/HomeScreenLayoutVertical.cpp

```diff
@@ -0,0 +1,176 @@
+            alarmTopIcon->setStatus(AlarmIcon::Status::ACTIVATED);
+            setScreenMode(ScreenMode::Main);
+            break;
+        case app::home_screen::ViewState::AlarmRinging:
+            setScreenMode(ScreenMode::Main);
+            alarmMainIcon->setStatus(AlarmIcon::Status::RINGING);
+            break;
+        case app::home_screen::ViewState::AlarmRingingDeactivatedWait:
+            setScreenMode(ScreenMode::AlarmActivatedDeactivated);
+            break;
+        case app::home_screen::ViewState::AlarmSnoozedWait:
+            alarmMainIcon->setStatus(AlarmIcon::Status::SNOOZE);
+            alarmActivatedDeactivatedScreen->image->set("big_alarm_snoozed_W_M", {});
+            setScreenMode(ScreenMode::AlarmActivatedDeactivated);
+            break;
+        case app::home_screen::ViewState::AlarmSnoozed:
+            alarmMainIcon->setStatus(AlarmIcon::Status::SNOOZE);
+            setScreenMode(ScreenMode::Main);
+            break;
+        }
+    }
+
+    auto HomeScreenLayoutVertical::setScreenMode(ScreenMode mode) -> void
+    {
+        switch (mode) {
+        case ScreenMode::Main:
+            mainScreen->setVisible(true);
+            setAlarmScreen->setVisible(false);
+            alarmActivatedDeactivatedScreen->setVisible(false);
+            mainScreen->informContentChanged();
+            break;
+        case ScreenMode::AlarmEdit:
+            mainScreen->setVisible(false);
+            setAlarmScreen->setVisible(true);
+            alarmActivatedDeactivatedScreen->setVisible(false);
+            setAlarmScreen->informContentChanged();
+            // This is an ugly hack to update format on alarm spinner.
+            // Will not be neccessary when visible flags get fixed
+            setAlarmFmtSpinner->setTimeFormat(setAlarmFmtSpinner->getTimeFormat());
+            break;
+        case ScreenMode::AlarmActivatedDeactivated:
+            mainScreen->setVisible(false);
+            setAlarmScreen->setVisible(false);
+            alarmActivatedDeactivatedScreen->setVisible(true);
+            alarmActivatedDeactivatedScreen->informContentChanged();
+            break;
+        }
+        resizeItems();
+    }
+
+    void HomeScreenLayoutVertical::setTemperature(utils::temperature::Temperature newTemp)
+    {}
+
+    void HomeScreenLayoutVertical::setTextDescription(const UTF8 &desc)
+    {
+        alarmActivatedDeactivatedScreen->text->setRichText(desc);
+        alarmActivatedDeactivatedScreen->text->informContentChanged();
+    }
+
+    bool HomeScreenLayoutVertical::isBatteryVisibilityAllowed(const Store::Battery &batteryContext)
+    {
+        return (batteryContext.level < battery_vertical::dischargingLevelShowTop) ||
+               (batteryContext.state == Store::Battery::State::Charging &&
+                batteryContext.level != battery_vertical::chargingLevelHideBot);
+    }
+
+    void HomeScreenLayoutVertical::setBatteryLevelState(const Store::Battery &batteryContext)
```

Comment:
Implementation here is identical with the `HomeScreenLayoutClassic::setBatteryLevelState`. The same goes for many other methods, `isBatteryVisibilityAllowed`